wasi/wasi.go

```diff
@@ -1359,10 +1360,16 @@ func newSnapshotPreview1(ctx context.Context) *snapshotPreview1 {
 	return &snapshotPreview1{sys: &defaultSys{}}
 }
 
-func sysCtx(m api.Module) *wasm.SysContext {
+func sysCtx(ctx context.Context, m api.Module) *wasm.SysContext {
 	if internal, ok := m.(*wasm.CallContext); !ok {
 		panic(fmt.Errorf("unsupported wasm.Module implementation: %v", m))
 	} else {
+		// Override FSContext when it is passed via context
+		if fsCtx := ctx.Value(expfs.FSKey{}); fsCtx != nil {
+			sys := *internal.Sys // copy
```

Comment:
hmm I don't think we want to do any copies here. maybe make an `fsCtx` function and use it where FS ops are used instead? you can leave this one alone. It is ok to call both sysCtx and also fsCtx in the same function body.. you could also make a special case that returns both, which can re-use the panic check and reduce the lookup overhead.

Reply:
I separated `FSContext` from `SysContext` and define a function returning both as you suggested. It affects many places as expected, but hopefully I understand what you meant.
https://github.com/tetratelabs/wazero/pull/571/commits/5ec6e1ac97c74cf00860df3f6d74871442421f14